### ISSUE TITLE
Missing namespace folders when using generators inside an engine

### ISSUE THREAD

**Issue by chancancode** [rails/rails#28010]
Rails version: 5.0.1

### Steps to reproduce

Generate an engine with >1 levels nested namespaces, e.g.

```bash
$ rails plugin new foo-bar --mountable
      create
      create  README.md
      create  Rakefile
      create  foo-bar.gemspec
      create  MIT-LICENSE
      create  .gitignore
      create  Gemfile
      create  app
      create  app/controllers/foo/bar/application_controller.rb
      create  app/helpers/foo/bar/application_helper.rb
      create  app/jobs/foo/bar/application_job.rb
      create  app/mailers/foo/bar/application_mailer.rb
      create  app/models/foo/bar/application_record.rb
      create  app/views/layouts/foo/bar/application.html.erb
      create  app/assets/images/foo/bar
      create  app/assets/images/foo/bar/.keep
      create  config/routes.rb
      create  lib/foo/bar.rb
      create  lib/tasks/foo/bar_tasks.rake
      create  lib/foo/bar/version.rb
      create  lib/foo/bar/engine.rb
      create  app/assets/config/foo_bar_manifest.js
      create  app/assets/stylesheets/foo/bar/application.css
      create  app/assets/javascripts/foo/bar/application.js
      create  bin/rails
      create  test/test_helper.rb
      create  test/foo/bar_test.rb
      append  Rakefile
      create  test/integration/navigation_test.rb
  vendor_app  test/dummy
```

That works as expected. However, when you run a generator inside this folder, it doesn't nest everything correctly, for example:

```
$ rails generate controller dashboard
      create  app/controllers/foo/dashboard_controller.rb
      invoke  erb
      create    app/views/foo/dashboard
      invoke  test_unit
      create    test/controllers/foo/dashboard_controller_test.rb
      invoke  helper
      create    app/helpers/foo/dashboard_helper.rb
      invoke    test_unit
      invoke  assets
      invoke    js
      create      app/assets/javascripts/foo/dashboard.js
      invoke    css
      create      app/assets/stylesheets/foo/dashboard.css
```

As you can see, it is supposed to generate `app/controllers/foo/bar/dashboard_controller.rb`, but it dropped the "bar" from the generated output.

**Comment by chancancode** [comment]
Related: https://github.com/rails/rails/pull/27550/files#r94523445

**Comment by chancancode** [comment]
Fixed by #27605
